### PR TITLE
V1 - Bug fix - Build-info link is wrong when used with projects

### PR DESCRIPTION
**Pull request by eyalbe4**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----
This PR fixes an issue for V1 and is a duplicate of https://github.com/jfrog/jfrog-cli-core/pull/216, which has been merged to V2 a few months ago.
It fixes an issue, whereby the build-info link displayed in the log when the build-info is published, is incorrect, when the build-info is published as part of a project.
This PR depends on https://github.com/jfrog/jfrog-client-go/pull/513

### CODE REVIEW

File: artifactory/commands/buildinfo/publish.go [hunk 1]
```diff
@@ -91,12 +96,50 @@ func (bpc *BuildPublishCommand) Run() error {
 	if err != nil {
 		return err
 	}
+
+	buildLink, err := bpc.constructBuildInfoUiUrl(servicesManager, buildInfo.Started)
```
Comment (RobiNino) [inline artifactory/commands/buildinfo/publish.go:100]: I suggest returning before this construction if on `dryRun`. The preview build was already logged to the terminal by the client, and a link was not printed prior to these changes.

File: utils/lock/utils_unix.go [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//go:build linux || darwin || freebsd
```
Comment (RobiNino) [inline utils/lock/utils_unix.go:1]: Is this relevant to the PR?

**Review by RobiNino — CHANGES_REQUESTED** [review]

File: utils/lock/utils_unix.go (hunk 2 above)
Comment (eyalbe4) [reply utils/lock/utils_unix.go:1]: It is not, but this is something go 1.17 automatically modifies. It did the same for v2 as well, after we started releasing it with go 1.17.
I'll make sure that v1 is also moved to go 1.17.

File: artifactory/commands/buildinfo/publish.go (hunk 1 above)
Comment (eyalbe4) [reply artifactory/commands/buildinfo/publish.go:100]: Good point @RobiNino. I'll add this change to v2 as well.
